tests/annotations_test.go

```diff
@@ -0,0 +1,45 @@
+package tests
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+// ensures services.annotations values are applied to both coderd deployment &
+// service
+func TestAnnotations(t *testing.T) {
+	t.Parallel()
+
+	chart := LoadChart(t)
+
+	expected := map[string]string{
+		"key": "value",
+		"service.beta.kubernetes.io/aws-load-balancer-backend-protocol": "http",
+	}
+
+	objs := chart.MustRender(t, func(cv *CoderValues) {
+		cv.Services.Annotations = expected
+	})
+
+	depl := MustFindDeployment(t, objs, "coderd")
```

Comment:
based on the docs, it looks like services.annotations applies the annotations to all services, whereas coderd.serviceAnnotations only applied the annotations to the coderd service. I'm not sure what the desired behavior should be here, actually. Looks like @deansheather added this in https://github.com/coder/enterprise-helm/pull/134 so maybe he can comment. Maybe the correct solution here is to bring back Dean's change, and use coderd.serviceAnnotations to control annotations on coderd only?

As a user, I think that I'd expect `service.annotations` to apply to all services (`timescale`, `coderd`) and `coderd.serviceAnnotations` only to apply to coderd. But it seems the previous behavior didn't apply `service.annotations` to the coderd service.

Reply:
You removed `coderd.serviceAnnotations` in #169

Reply:
yea, that was unintentional though. I'm not sure what the desired behavior should be here, because:

1. the previous behavior (prior to my change) was that services.annotations was used for ?? and did not apply to coderd (I would have expected it'd apply to coderd too, as well as other services we create)
2. the previous behavior (prior to my change) was that coderd.serviceAnnotations created annotations that apply to the coderd service
3. the current behavior (after my change) applies services.annotations to coderd and other services, but of course, now coderd.serviceAnnotations doesn't do anything
   https://github.com/coder/enterprise-helm/blob/8aef31f7a0c135a073bf9be75523e7fad55b3a49/templates/coderd.yaml#L240-L256

   I'm not sure why we're applying the annotations to the StatefulSet, rather than the timescale Service:

   https://github.com/coder/enterprise-helm/blob/0da85a6f4adc20f9e418e8f8da29ce9a978dc7c3/templates/timescale.yaml#L24-L36

   and I don't know why we're applying this to coderd's deployment:

   https://github.com/coder/enterprise-helm/blob/0da85a6f4adc20f9e418e8f8da29ce9a978dc7c3/templates/coderd.yaml#L2-L16

@deansheather so my question for you is whether we need to be able to control service annotations on a service-by-service basis (e.g. the timescale annotations vs the coderd annotations), in which case the current behavior is broken (we can set annotations for all services but not for coderd only)

Maybe the naming was confusing - does "services" refer to k8s services, or is it referring to coderd/timescale collectively as "services"? The code seems to do the latter, in which case the `coderd.serviceAnnotations` thing makes sense

Reply:
I'm unsure but it's better to have annotations for all individual components be separate values IMO. So `coderd.annotations` `coderd.service(\.a|A)nnotations` `timescale.annotations`

Reply:
If we go with such a solution, that would require removing `services.annotations`, thus rendering this test null. I'll hold off merging this until we decide.